reconcile/test/external_resources/test_module_configuration.py

```diff
@@ -1,48 +1,132 @@
+                resource={},
+                namespace={},
+            ),
+            None,
+            "candidate-image",
+            "2.0.0",
         ),
+        (  # Module Overrides image/version
+            ExternalResourceSpec(
+                provision_provider="aws",
+                provisioner={"external_resources": {"channel": "candidate"}},
+                resource={"channel": "experiment"},
+                namespace={},
+            ),
+            ExternalResourcesModuleOverrides(
+                image="overridden-image",
+                version="overridden-version",
+                module_type=None,
+                channel=None,
+                reconcile_timeout_minutes=None,
+                outputs_secret_image="overridden-secrets-image",
+                outputs_secret_version="overridden-secrets-version",
+                resources=DeployResourcesFields(
+                    requests=ResourceRequestsRequirementsV1(cpu="100m", memory="128Mi"),
+                    limits=ResourceLimitsRequirementsV1(memory="4Gi", cpu=None),
+                ),
+            ),
+            "overridden-image",
+            "overridden-version",
+        ),
+        (  # Module Overrides channel
+            ExternalResourceSpec(
+                provision_provider="aws",
+                provisioner={"external_resources": {"channel": "candidate"}},
+                resource={"channel": "experiment"},
+                namespace={},
+            ),
+            ExternalResourcesModuleOverrides(
+                image=None,
+                version=None,
+                module_type=None,
+                channel="experiment-2",
+                reconcile_timeout_minutes=None,
+                outputs_secret_image="overridden-secrets-image",
+                outputs_secret_version="overridden-secrets-version",
+                resources=DeployResourcesFields(
+                    requests=ResourceRequestsRequirementsV1(cpu="100m", memory="128Mi"),
+                    limits=ResourceLimitsRequirementsV1(memory="4Gi", cpu=None),
+                ),
+            ),
+            "experiment-2-image",
+            "4.0.0",
+        ),
+    ],
+)
+def test_module_image_configuration(
+    module: ExternalResourcesModuleV1,
+    settings: ExternalResourcesSettingsV1,
+    spec: ExternalResourceSpec,
+    overrides: ExternalResourcesModuleOverrides | None,
+    expected_image: str,
+    expected_version: str,
+) -> None:
+    if overrides:
+        spec.metadata = {"module_overrides": overrides}
+
+    conf = ExternalResourceModuleConfiguration.resolve_configuration(
+        module=module, spec=spec, settings=settings
     )
+    assert conf.image == expected_image
+    assert conf.version == expected_version
+    if overrides:
+        assert conf.outputs_secret_image == overrides.outputs_secret_image
+        assert conf.outputs_secret_version == overrides.outputs_secret_version
+        assert conf.resources == overrides.resources
```

Comment:
`if` condition should be avoided in tests, can we include `metadata` inside params and assert on full `conf` object?

Reply:
Metadata is no part of the constructor. I added the whole configuration object. PTAL.